src/library_pthread.js

```diff
@@ -263,9 +263,9 @@ var LibraryPThread = {
           return;
         }
 
-        if (cmd === 'processQueuedMainThreadWork') {
+        if (cmd === 'processProxyingQueue') {
           // TODO: Must post message to main Emscripten thread in PROXY_TO_WORKER mode.
-          _emscripten_main_thread_process_queued_calls();
+          _emscripten_proxy_execute_queue(d['queue']);
```

Comment:
Could we save a little of JS codesize here by passing zero args and having `_emscripten_proxy_execute_queue` assume that NULL == system queue?  Would save the extra post message packing / unpacking too.

Reply:
Yes, potentially, but I would prefer to make it a clear-cut error for NULL to ever be passed to `emscripten_proxy_execute_queue`. Better for users to get an assertion failure (in debug builds) when they do that by accident than for the system queue to be executed.